### PR TITLE
[DOCS] Improves description for forecast_stats

### PR DESCRIPTION
**Pull request by szabosteve**
Improves the description of the `forecast_stats` objects of the GET job stats API docs.

Related issue: https://github.com/elastic/ml-team/issues/274#issuecomment-571714977

Preview: http://elasticsearch_50729.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-get-job-stats.html

### CODE REVIEW

File: docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc [hunk 1]
```diff
@@ -158,24 +159,28 @@ NOTE: Unless there is at least one forecast, `memory_bytes`, `records`,
...
 
 `forecasts_stats`.`forecasted_jobs`:::
-(long) The number of jobs that have at least one forecast.
+(long) A value of `0` indicates that forecasts do not exist for this job. A 
+value of `1` indicates that at least one forecast exists.
 
 `forecasts_stats`.`memory_bytes`:::
-(object) Statistics about the memory usage: minimum, maximum, average and total.
+(object) The `avg`, `min`, `max` and `total` memory usage in bytes for forecasts 
+related to this job. Omitted if forecasts do not exist.
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc]: I generally like starting the sentence with the conditional part, so folks can stop reading if it doesn't apply to them.
```suggestion
related to this job. If there are no forecasts, this property is omitted.
```

File: docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc [hunk 2]
```diff
@@ -158,24 +159,28 @@ NOTE: Unless there is at least one forecast, `memory_bytes`, `records`,
...
 
 `forecasts_stats`.`records`:::
-(object) Statistics about the number of forecast records: minimum, maximum,
-average and total.
+(object) The `avg`, `min`, `max` and `total` number of model_forecast documents 
+written for forecasts related to this job. Omitted if forecasts do not exist.
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc]:
```suggestion
written for forecasts related to this job. If there are no forecasts, this property is omitted.
```

File: docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc [hunk 3]
```diff
@@ -158,24 +159,28 @@ NOTE: Unless there is at least one forecast, `memory_bytes`, `records`,
...
 
 `forecasts_stats`.`processing_time_ms`:::
-(object) Statistics about the forecast runtime in milliseconds: minimum, maximum,
-average and total.
+(object) The `avg`, `min`, `max` and `total` runtime in milliseconds for 
+forecasts related to this job. Omitted if forecasts do not exist.
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc]:
```suggestion
forecasts related to this job. If there are no forecasts, this property is omitted.
```

File: docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc [hunk 4]
```diff
@@ -158,24 +159,28 @@ NOTE: Unless there is at least one forecast, `memory_bytes`, `records`,
...
 
 `forecasts_stats`.`status`:::
-(object) Counts per forecast status. For example: `{"finished" : 2}`. 
+(object) The count of forecasts by their status. For example: 
+{"finished" : 2, "started" : 1}. Omitted if forecasts do not exist.
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc]:
```suggestion
{"finished" : 2, "started" : 1}. If there are no forecasts, this property is omitted.
```

**Review by lcawl — APPROVED** [review]
Added some suggestions, but otherwise LGTM!

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: d7bb5d7
7.5: a4f42c5

**Pull request merged**